### PR TITLE
web: highlight search operators and/or/not

### PR DESCRIPTION
**Pull request by rvantonder**
Adds highlighting for and/or/not. It does _not_ have to be these colors. @rrhyne @felixfbecker please weigh in your thoughts on a color choice for operators and I will update it. I picked the dark mode purple-ish color based on the VScode rendering of the `return` keyword in vscode, (which I think uses our theme?). For light mode I just picked something reasonably close.

Here is what it currently looks like:

Screen Shot 2020-08-03 at 11 29 42 PM

Screen Shot 2020-08-03 at 11 37 54 PM

Since I didn't find tests that would specifically break the color highlighting, I didn't add any for that. ~Though, I'll add some tests to make sure we lex the right ranges for these operators~ I did some exhaustive manual testing, and also updated our existing tests that already have test inputs for parsing operators.

### CODE REVIEW

File: shared/src/search/parser/parser.test.ts [hunk 1]
```diff
@@ -609,8 +609,7 @@ describe('parseSearchQuery()', () => {
                             start: 15,
                         },
                         token: {
-                            type: 'literal',
-                            value: 'and',
+                            type: 'operator',
```
Comment (rvantonder) [inline shared/src/search/parser/parser.test.ts:612]: Think we probably want to preserve the 'value' of these operators in the long run, but they are not significant for highlighting. I didn't see an obvious way to modify the `pattern` parser to preserve the value (i.e., without duplicating that function). I may go back and try to improve this, but am just punting for now.

Comment (lguychard) [reply shared/src/search/parser/parser.test.ts:612]: Pushed https://github.com/sourcegraph/sourcegraph/pull/12694/commits/666ed5a85fd27e0e7c221b917abc57f01acdc855 to help out with this

**Comment by rvantonder** [comment]
@AlicjaSuska happy to get your thoughts on a color choice for these `AND`/`or`/`not` operators in our query that aligns with our current colors.

**Comment by felixfbecker** [comment]
> @AlicjaSuska happy to get your thoughts on a color choice for these AND/or/not operators in our query that aligns with our current colors.

The best way to make sure the colors a part of our color palette (OpenColor) would be to import them instead of using hex codes. You can pick colors from here: https://yeun.github.io/open-color/
And import them like this:
```ts
import openColor from 'open-color'

// example usage
openColor.blue[7]
```

(but I'd also love to hear the @sourcegraph/design team's thoughts on which color and shade would be best)

File: shared/src/search/parser/parser.test.ts (hunk 1 above)
Comment (rvantonder) [reply shared/src/search/parser/parser.test.ts:612]: @lguychard we need to do more web coding together!!1

**Comment by rrhyne** [comment]
@rvantonder, I think the choice of purple is perfect. The only change I would make is to select two similar colors from the open color palette as Felix mentioned: Here is a comp showing the two I picked:

https://www.figma.com/file/DzmwBVeEu4cADYIoz5EefH/%2312694-and%2For%2Fnot?node-id=0%3A1&viewport=1298%2C1014%2C1

#CC5DE8 (#5 of 10 purple) almost works for both, but it's a tad light on the white background.

**Comment by rvantonder** [comment]
@rrhyne me gusta. Going with that.

File: shared/src/search/parser/parser.test.ts (hunk 1 above)
Comment (lguychard) [reply shared/src/search/parser/parser.test.ts:612]: I'd love that! I'll trade you web coding sessions for backend parser sessions 😄

**Pull request merged**